source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -224,6 +227,16 @@ void AuthenticatorImpl::verifyKey() {
     // Remove JWT from headers.
     curr_token_->removeJwt(*headers_);
   }
+  if (set_payload_cb_ && !provider.payload_in_metadata().empty()) {
+    try {
+      ProtobufWkt::Struct payload_pb;
+      MessageUtil::loadFromJson(jwt_->payload_str_, payload_pb);
+      set_payload_cb_(provider.payload_in_metadata(), payload_pb);
+    } catch (EnvoyException ex) {
```

Comment:
In general, you can't throw exceptions on the data path..

Reply:
I don't think we have policy/style guid not throw exception on data path. It's true in general envoy core doesn't catch in data path exceptions, but as long as it is caught by filter itself it should be OK? I know some extensions does this a lot (e.g thrift proxy).

Reply:
If the filter catches it's OK, but it does make reasoning about behavior harder. It would significantly simplify the understandability of exception safety to readers to avoid this.

Reply:
Use RELEASE_ASSERT to replace this try and catch.   The error should not happen.